include/kernel.h

```diff
@@ -3925,7 +3949,7 @@ extern void _timer_expiration_handler(struct _timeout *t);
  * @param sym Declared stack symbol name
  * @return The buffer itself, a char *
  */
-#define K_THREAD_STACK_BUFFER(sym) sym
+#define K_THREAD_STACK_BUFFER(sym) ((char *) sym)
```

Comment:
Is this macro supposed to be used only within function contexts? If so, might consider having a `static inline char *k_thread_stack_buffer(k_thread_stack_t sym)` function instead for a little bit more type checking.

An alternative would be to expand this to a compound statement instead, with a call to `BUILD_ASSERT()`, although this would require the usage of a GCC extension (`__builtin_types_compatible_p()`, which Clang also provides). Nontested:

    #define K_THREAD_STACK_BUFFER(sym) \
        ({ \
                BUILD_ASSERT(__builtin_types_compatible_p(typeof(sym), k_thread_stack_t)); \
                ((char *)sym) \
        })